src/QueryEditor.tsx

```diff
@@ -13,6 +13,20 @@ type Props = QueryEditorProps<DataSource, RedshiftQuery, RedshiftDataSourceOptio
 type QueryProperties = 'schema' | 'table' | 'column';
 
 export function QueryEditor(props: Props) {
+  const state = props.data?.state;
+  const [running, setRunning] = useState(false);
+  const [stopping, setStopping] = useState(false);
+  const [lastState, setLastState] = useState(state);
+
+  useEffect(() => {
+    if (state && lastState !== state && state !== LoadingState.Loading) {
+      setRunning(false);
+      setStopping(false);
+    }
+
+    setLastState(state);
```

Comment:
I'm guessing state is an object right? Does this update the state perhaps too many times? Like if we rerender the component for another reason and send it a new copy of the object wouldn't it rerender a lot more?

Reply:
It's a `string | undefined`. [LoadingState](https://github.com/grafana/grafana/blob/3ea9ece16e1172d83ae7c94dfe8b63cfb2c20451/packages/grafana-data/src/types/data.ts#L15-L21) is an enum.